server/libs/infra/src/infra.config.ts

```diff
@@ -26,17 +29,24 @@ function parseRedisConfig(): RedisOptions {
 
 export const redisConfig: RedisOptions = parseRedisConfig();
 
-export const bullConfig: BullModuleOptions = {
+interface CustomBullModuleOptions extends BullModuleOptions {
+  onError?: (error: any) => void;
+}
+
+export const bullConfig: CustomBullModuleOptions = {
   prefix: 'immich_bull',
   redis: redisConfig,
   defaultJobOptions: {
     attempts: 3,
     removeOnComplete: true,
     removeOnFail: false,
   },
+  onError: (error) => {
+    logger.error(error);
+  },
 };
 
-export const bullQueues: BullModuleOptions[] = Object.values(QueueName).map((name) => ({ name }));
+export const bullQueues: CustomBullModuleOptions[] = Object.values(QueueName).map((name) => ({ name }));
```

Comment:
Is this change necessary?

Reply:
I believe so as the BullModuleOptions doesn't expose the onError, but I could be mistaken too.

Reply:
This is only the config for the queues for not main config.

Reply:
Let me change it back as the main config should be enough.